src/awe/rmdparchive.cpp

```diff
@@ -289,143 +282,93 @@ void RMDPArchive::loadHeaderV2(Common::ReadStream *bin) {
 	_folderEntries.resize(numFolders);
 	_fileEntries.resize(numFiles);
 
-	for (auto &entry : _folderEntries) {
-		entry.nameHash = bin->readUint32BE();
-		entry.nextNeighbourFolder = bin->readUint32BE();
-		entry.prevFolder = bin->readUint32BE();
-
-		bin->skip(4); // Unknown value
+	for (auto &entry : _folderEntries)
+		entry = this->readFolder(bin, end, nameSize);
```

Comment:
I have consulted my reversing work for the rmdp format, and found out that the folder data block changed in quantum break. The properties nextNeighbourFolder and prevFolder were extended to 64 bit values. That was my error when I first wrote the v8 header loading. From this you cannot assume the folder entries to be uniform.

Reply:
That... Actually changes a lot, because it not only makes folder entries different but also changes the FolderEntry structure and the way some of the checks should be handled. I'll try to address that.